create.ps1

```diff
@@ -1,234 +1,356 @@
+        $ConnectionString,
+
+        [parameter(Mandatory = $false)]
+        $Username,
+
+        [parameter(Mandatory = $false)]
+        $Password,
+
+        [parameter(Mandatory = $true)]
+        $SqlQuery,
+
+        [parameter(Mandatory = $true)]
+        [ref]$Data
+    )
+    try {
+        $Data.value = $null
+
+        # Initialize connection and execute query
+        if (-not[String]::IsNullOrEmpty($Username) -and -not[String]::IsNullOrEmpty($Password)) {
+            # First create the PSCredential object
+            $securePassword = ConvertTo-SecureString -String $Password -AsPlainText -Force
+            $credential = [System.Management.Automation.PSCredential]::new($Username, $securePassword)
+
+            # Set the password as read only
+            $credential.Password.MakeReadOnly()
+
+            # Create the SqlCredential object
+            $sqlCredential = [System.Data.SqlClient.SqlCredential]::new($credential.username, $credential.password)
         }
-    }
-    $accountExists.Close()
 
-    # Check if user already exists in the Salto Database
-    write-verbose -Verbose -Message "$lookupResult"
-    Switch ($lookupResult.count) {
-        0 {
-            Write-Verbose -Verbose -Message "Account record does not exist. Creating account record for person '$($p.displayName)' with ExtUserId '$($account.ExtUserId)'"
+        # Connect to the SQL server
+        $SqlConnection = [System.Data.SqlClient.SqlConnection]::new()
+        $SqlConnection.ConnectionString = “$ConnectionString”
+        if (-not[String]::IsNullOrEmpty($sqlCredential)) {
+            $SqlConnection.Credential = $sqlCredential
+        }
+        $SqlConnection.Open()
+        Write-Verbose "Successfully connected to SQL database" 
+
+        # Set the query
+        $SqlCmd = [System.Data.SqlClient.SqlCommand]::new()
+        $SqlCmd.Connection = $SqlConnection
+        $SqlCmd.CommandText = $SqlQuery
+
+        # Set the data adapter
+        $SqlAdapter = [System.Data.SqlClient.SqlDataAdapter]::new()
+        $SqlAdapter.SelectCommand = $SqlCmd
+
+        # Set the output with returned data
+        $DataSet = [System.Data.DataSet]::new()
+        $null = $SqlAdapter.Fill($DataSet)
+
+        # Set the output with returned data
+        $Data.value = $DataSet.Tables[0] | Select-Object -Property * -ExcludeProperty RowError, RowState, Table, ItemArray, HasErrors
+    }
+    catch {
+        $Data.Value = $null
+        Write-Error $_
+    }
+    finally {
+        if ($SqlConnection.State -eq "Open") {
+            $SqlConnection.close()
         }
-        1 {
-            $account.ExtUserID = $lookupResult[0]
-            Write-Verbose -Verbose -Message "Account record exists in the Salto Space database. Correlating account record for person '$($p.displayName)' with ExtUserId '$($account.ExtUserId)'"
+        Write-Verbose "Successfully disconnected from SQL database"
+    }
+}
+
+function ConvertTo-FlatObject {
```

Comment:
Do we still need this function now the fields with dots in it are gone in the field mapping?